base/shared/src/main/scala/scalaz/ct/profunctor.scala

```diff
@@ -7,6 +7,10 @@ trait ProfunctorClass[F[_, _]] {
   def lmap[A, B, C](fab: F[A, B])(ca: C => A): F[C, B]
   def rmap[A, B, C](fab: F[A, B])(bc: B => C): F[A, C]
   def dimap[A, B, C, D](fab: F[A, B])(ca: C => A)(bd: B => D): F[C, D]
+  def functor[A]: Functor[F[A, ?]] =
+    instanceOf(new FunctorClass[F[A, ?]] {
+      def map[B, C](fab: F[A, B])(f: B => C): F[A, C] = rmap(fab)(f)
+    })
```

Comment:
not sure exactly how much I like having this as an instance method.

Reply:
It should probably be moved into a RightFunctor class, which will be a superclass of both Profunctor and Bifunctor

Reply:
for this PR, I think I'd prefer have this implementation directly in `BaseHierarchy` and this method removed from `[Pro/Bi]functor`.

Reply:
done